packages/component-library/src/components/feedback-indicator/mt-toast/mt-toast.vue

```diff
@@ -0,0 +1,245 @@
+<template>
+  <div class="mt-toast" :class="classes" @mouseenter="onMouseEnter" @mouseleave="onMouseLeave">
+    <!-- This is here for layout reasons -->
+    <div class="mt-toast__bottom-left"></div>
+
+    <div class="mt-toast__bottom-center">
+      <Transition name="quick-toasts">
+        <mt-toast-notification
+          v-if="centerToast"
+          :toast="centerToast"
+          @remove-toast="$emit('remove-toast', centerToast.id)"
+          :quick-display="true"
+          :key="centerToast.id"
+        />
+      </Transition>
+    </div>
+
+    <div class="mt-toast__bottom-right">
+      <TransitionGroup name="toasts">
+        <mt-toast-notification
+          v-for="(toast, index) in rightToasts"
+          :key="toast.id"
+          :toast="toast"
+          @remove-toast="$emit('remove-toast', toast.id)"
+          :style="{ '--num': index }"
+          :index="index"
+        />
+      </TransitionGroup>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { type PropType, ref, toRef, defineProps, computed, watch, defineEmits } from "vue";
+import MtToastNotification from "./mt-toast-notification.vue";
+
+export interface Toast {
+  id: number;
+
+  // Public api
+  msg: string; // Toast message
+  type: "informal" | "critical" | "positive"; // Toast type
+  icon?: string; // An icon to be displayed before the message
+  // Adds a button to the toast
+  action?: {
+    label: string; // Label of the button
+    callback: () => void; // Callback will be called on click
+  };
+  dismissible?: boolean; // Toggles if the toast is manually closable
+}
+
+const emit = defineEmits(["remove-toast"]);
+
+const props = defineProps({
+  /**
+   * A collection of toasts. The most current one needs to be prepended.
+   */
+  toasts: {
+    type: Array as PropType<Toast[]>,
+    required: true,
+  },
+});
+
+const isHovered = ref(false);
+const toasts = toRef(() => props.toasts);
+const hoverTimeoutId = ref<number | undefined>(undefined);
+
+const centerToast = computed(() => {
+  const ct = toasts.value.filter((t) => {
+    return !t.action && t.type !== "critical" && !t.dismissible;
+  });
+
+  return ct[0];
+});
+
+const rightToasts = computed(() => {
+  return toasts.value.filter((t) => {
+    return t.action || t.type === "critical" || t.dismissible;
+  });
+});
+
+const classes = computed(() => {
+  return {
+    "mt-toast__collapsed": !isHovered.value,
+    "mt-toast__expanded": isHovered.value,
+  };
+});
+
+watch(centerToast, (_, oldToast) => {
+  if (!oldToast) {
+    return;
+  }
+
+  emit("remove-toast", oldToast.id);
+});
+
+function onMouseEnter() {
+  if (hoverTimeoutId.value) {
+    window.clearTimeout(hoverTimeoutId.value);
+  }
+
+  isHovered.value = true;
+}
+
+function onMouseLeave() {
+  hoverTimeoutId.value = window.setTimeout(() => {
+    isHovered.value = false;
+  }, 500);
+}
+</script>
+
+<style lang="scss">
```

Comment:
Can we used scoped styles here?

Reply:
It's not entirely possible to scope all the styling since we change childs of mt-toast-notification. But I extracted that part and made the rest scoped.